### ISSUE TITLE
在opengait上使用单卡训练步态识别的模型时，得到的准确率全是2%，这是怎么回事呢？

### ISSUE THREAD

**Issue by heyihao2018** [ShiqiYu/OpenGait#234]
大家好，在我复现opengait的gaitset代码时，因为我用的是单卡，所以我把分布式相关的代码都注释掉了，我个人觉得训练过程应该是没有问题的，最可能出现问题的是我用的windows系统，这是否会有影响，以及测试的代码是否存在小问题，如果有人也出现过这个问题，请大家告诉我，我自己也会慢慢琢磨的，谢谢大家！

**Comment by world1987** [comment]
和系统关系不大，win和linux都复现过，单卡需要调整参数

**Comment by wj1tr0y** [comment]
单卡可以不用改动代码，直接改动train和test的命令即可，CUDA_VISBLE_DEVICES改成0，nproc_per_node改成1

**Comment by JJ-Luov** [comment]
你好，请问这个问题你解决了吗

**Comment by heyihao2018** [comment]
`def ddp_all_gather(features, dim=0, requires_grad=False):
    '''
        inputs: [n, ...]
    '''

    world_size = 1
    rank = 0
    feature_list = [features for _ in range(world_size)]
    if requires_grad:
        feature_list[rank] = features
    feature = torch.cat(feature_list, dim=dim)
    return feature`

我把lib/utils/common.py文件里面的这个函数改了改，就可以了

**Comment by heyihao2018** [comment]
你好，我在gaithub里面已经回复你了




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年9月12日(星期四) 下午4:44
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ShiqiYu/OpenGait] 在opengait上使用单卡训练步态识别的模型时，得到的准确率全是2%，这是怎么回事呢？ (Issue #234)














 
你好，请问这个问题你解决了吗

**Comment by yyj-001** [comment]
您好，请问您的问题解决了吗？我也遇到了类似的问题，想参考前辈。我在Linux系统上用单卡和双卡都试过了，训练结果都只有2%左右的rank1，我没有改动代码，只改动了.yaml文件里的参数。目前我找不到问题在哪里了

**Comment by JJ-Luov** [comment]
你好，我把网络换成gaitset能收敛，10000次迭代rank1能到百分之九十多，具体也不知道是哪里的问题
&nbsp;


J.&nbsp;X&nbsp;J.
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年11月11日(星期一) 下午5:22
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ShiqiYu/OpenGait] 在opengait上使用单卡训练步态识别的模型时，得到的准确率全是2%，这是怎么回事呢？ (Issue #234)














 
您好，请问您的问题解决了吗？我也遇到了类似的问题，想参考前辈。我在Linux系统上用单卡和双卡都试过了，训练结果都只有2%左右的rank1，我没有改动代码，只改动了.yaml文件里的参数。目前我找不到问题在哪里了
